### PR TITLE
[Sprint: 42] XD-2628 Support for Retry in Redis Analytics operations

### PR DESCRIPTION
**Pull request by markpollack**

### CODE REVIEW

File: config/servers.yml [hunk 1]
```diff
@@ -337,3 +340,11 @@ endpoints:
...
 #    enabled: true
 ---
+
+#xd:
+#  analytics:
+#    redis:
+#         backOffInitialInterval:    1000
+#         backOffMaxInterval:        10000
+#         backOffMultiplier:         2.0
+#         maxAttempts:               5
```
Comment (markpollack) [inline config/servers.yml]: oops, please fix on merge.

File: config/xd-container-logger.properties [hunk 2]
```diff
@@ -22,7 +22,7 @@ log4j.logger.org.springframework.xd.sqoop=INFO
 log4j.logger.org.springframework=WARN
 log4j.logger.org.springframework.boot=WARN
 log4j.logger.org.springframework.integration=WARN
-log4j.logger.org.springframework.retry=WARN
+log4j.logger.org.springframework.retry=DEBUG
```
Comment (markpollack) [inline config/xd-container-logger.properties]: oops again, please switch back to WARN on merge.

File: spring-xd-analytics/src/test/java/org/springframework/xd/analytics/metrics/common/RedisRepositoriesConfig.java [hunk 3]
```diff
@@ -84,6 +90,18 @@ public StringRedisTemplate stringRedisTemplate() {
 	}
 
 	@Bean
+	public RetryOperations retryOperations() {
+		RetryTemplate retryTemplate = new RetryTemplate();
+		retryTemplate.setRetryPolicy(new SimpleRetryPolicy(3, Collections.<Class<? extends Throwable>, Boolean> singletonMap(RedisConnectionFailureException.class, true)));
+		ExponentialBackOffPolicy backOffPolicy = new ExponentialBackOffPolicy();
+		backOffPolicy.setInitialInterval(1000L);
+		backOffPolicy.setMaxInterval(1000L);
```
Comment (mbogoevici) [inline spring-xd-analytics/src/test/java/org/springframework/xd/analytics/metrics/common/RedisRepositoriesConfig.java]: Shouldn't this be `10000` ? The initial and max interval are the same right now.

Comment (markpollack) [inline spring-xd-analytics/src/test/java/org/springframework/xd/analytics/metrics/common/RedisRepositoriesConfig.java]: Can change it if you want, I don't think the tests that use this configuration exercise the retry

Comment (mbogoevici) [inline spring-xd-analytics/src/test/java/org/springframework/xd/analytics/metrics/common/RedisRepositoriesConfig.java]: d'oh. I didn't realize this was a test configuration only.

**Comment by mbogoevici** [comment]
Merged w/https://github.com/spring-projects/spring-xd/commit/d4503de090b44e69c8a5761a8656ecc39b748848

**Pull request closed without merging**